proposals/image-metadata.md

```diff
@@ -0,0 +1,81 @@
+# Image Metadata
+
+## Goal
+
+Record dev container config and feature metadata in prebuilt images, such that, the image and the built-in features can be used with a devcontainer.json (image-, Dockerfile- or Docker Compose-based) that does not repeat the dev container config or feature metadata. Other tools should be able to record the same metadata without necessarily using features themselves.
+
+Current dev container config that can be recorded in the image: ``
+
+Current feature metadata relevant for using the feature when it is already part of the image: `mounts`, `init`, `privileged`, `capAdd`, `securityOpt`, `entrypoint` and `customizations`.
```

Comment:
Is this trying to record current state?  I wasn't quite sure if this was trying to get at the relevant properties that would apply to image labels or just how things are now.

Reply:
Current state. Apparently forgot to paste in the list for the devcontainer.json above. (We could remove these as we repeat them in the table at the end.)

Reply:
Actually: "Current state" in the sense that these are the currently existing properties we will record in the new image metadata. :)

Reply:
Yeah that was kind of what I was thinking too - the table at the end covers it.